addons/account/models/account_move.py

```diff
@@ -3488,7 +3488,7 @@ class AccountMoveLine(models.Model):
         readonly=True, store=True,
         help='Utility field to express amount currency')
     account_id = fields.Many2one('account.account', string='Account',
-        index=True, ondelete="cascade",
+        index=True, ondelete="cascade", auto_join=True,
         domain="[('deprecated', '=', False), ('company_id', '=', 'company_id'),('is_off_balance', '=', False)]",
         check_company=True,
```

Comment:
Apparently, with `auto_join=True`, we should implement multi-company filter somewhere.

Reply:
So?

But doesn't the `check_company` do that?

Reply:
As I understand, `check_company=True` is ignored if we have `auto_join=True`.
I'll make further investigation and try to measure improvements

Reply:
Isn't it the record rules that are ignored? `check_company` has nothing to do with record rules